### PR TITLE
Refactor position folding

### PR DESCRIPTION
**Pull request by fweik**
Fixes #

Description of changes:
 - 

PR Checklist
------------
 - [ ] Tests?
   - [ ] Interface
   - [ ] Core 
 - [ ] Docs?

### CODE REVIEW

File: src/core/algorithm/periodic_fold.hpp [hunk 1]
```diff
@@ -0,0 +1,61 @@
+#ifndef CORE_ALGORITHM_PERIODIC_FOLD_HPP
+#define CORE_ALGORITHM_PERIODIC_FOLD_HPP
+
+#include <limits>
+#include <utility>
+#include <cmath>
+
+namespace Algorithm {
+    /**
+     * @brief Fold value in to primary interval.
```
Comment (KaiSzuttor) [inline src/core/algorithm/periodic_fold.hpp]: into

File: src/core/algorithm/periodic_fold.hpp [hunk 2]
```diff
@@ -0,0 +1,61 @@
...
+     *
+     * @param x Value to fold
+     * @param i Image cont befor folding
```
Comment (KaiSzuttor) [inline src/core/algorithm/periodic_fold.hpp]: count

Comment (KaiSzuttor) [reply src/core/algorithm/periodic_fold.hpp]: before

File: src/core/algorithm/periodic_fold.hpp [hunk 3]
```diff
@@ -0,0 +1,61 @@
...
+  while ((x >= l) && (i < limits::max())) {
+    x -= l;
+    ++i;
+  }
+
+  return {x, i};
+}
+
+    /**
+     * @brief Fold value in to primary interval.
```
Comment (KaiSzuttor) [inline src/core/algorithm/periodic_fold.hpp]: into

File: src/core/grid.cpp [hunk 4]
```diff
@@ -71,15 +71,10 @@ void init_node_grid() {
...
-  double f_pos[3];
+  auto const f_pos = folded_position(pos);
 
-  for (i = 0; i < 3; i++)
-    f_pos[i] = pos[i];
-
-  fold_position(f_pos, im);
-
-  for (i = 0; i < 3; i++) {
+  int im[3] = {0, 0, 0};
```
Comment (KaiSzuttor) [inline src/core/grid.cpp]: Vector3i

**Comment by fweik** [comment]
@KaiSzuttor

File: src/core/grid.hpp [hunk 5]
```diff
@@ -203,26 +205,21 @@ Vector3d get_mi_vector(T const &a, U const &b) {
 
 /** fold a coordinate to primary simulation box, including velocity.
```
Comment (KaiSzuttor) [inline src/core/grid.hpp]: remove "including velocity"

**Review by KaiSzuttor — APPROVED** [review]

**Pull request merged**